aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -114,6 +114,46 @@ static std::unique_ptr<TensorIterator> make_reduction(
   return TensorIterator::reduce_op(viewed_result, self.to(dtype));
 }
 
+static std::unique_ptr<TensorIterator> make_reduction(
+    const char* name, Tensor& result1, Tensor& result2, const Tensor& self, c10::optional<IntArrayRef> dim,
+    bool keepdim, ScalarType dtype)
+{
+  // check that result type and dtype match if provided
+  AT_CHECK(
+      !result1.defined() || result1.type().scalarType() == dtype,
+      name, ": provided dtype must match dtype of result. Got ",
+      toString(result1.type().scalarType()),
+      " and ",
+      toString(dtype),
+      ".");
+
+  AT_CHECK(
+      !result2.defined() || result2.type().scalarType() == dtype,
+      name, ": provided dtype must match dtype of result. Got ",
+      toString(result2.type().scalarType()),
+      " and ",
+      toString(dtype),
+      ".");
+
+  int64_t ndim = self.dim();
+  DimMask mask = make_dim_mask(dim, ndim);
+  allocate_reduction_result(result1, self, mask, keepdim, dtype);
+  auto viewed_result1 = review_reduce_result(result1, ndim, mask, keepdim);
+
+  allocate_reduction_result(result2, self, mask, keepdim, dtype);
+  auto viewed_result2 = review_reduce_result(result2, ndim, mask, keepdim);
+
+  // special case for type promotion in mixed precision, improves computational
+  // efficiency.
+  // not generalize this to common mismatched input/output types to avoid cross
```

Comment:
English grammar. "We don't generalize this to common mismatched input/output types to avoid..."

Reply:
Fixed